### DISCUSSION TITLE
augmenting both image and mask

### DISCUSSION THREAD

**Discussion by yeeseng** [Project-MONAI/MONAI#2779] [Q&A]
I'm confused on how to perform identical augmentation on the image/mask pair. 

For example, I have an augmentation routine that I can apply to a 3D volume. However, I'm unsure how I can apply the same augmentation to the accompanying mask for training a segmentation model.

```
def get_training_augmentation():
    augmentationTransform = monai.transforms.Compose(transforms=[monai.transforms.RandFlip(prob=0.5, spatial_axis=0),
                                                                 monai.transforms.RandAffine(prob=0.9,
                                                                                             rotate_range=math.pi,
                                                                                             shear_range=0.1,
                                                                                             translate_range=10,
                                                                                             scale_range=0.1),
                                                                 monai.transforms.RandGaussianNoise(prob=0.7, mean=0.0,
                                                                                                    std=0.0005)
                                                                 ],
                                                     map_items=True, unpack_items=False)
    return augmentationTransform
```
I saw a few examples in the tutorial that added "keys" as one of the argument into each transform like keys=['image', 'mask']. That didn't seem to work with the above transformations. And after that, I'm not sure how to apply the transformations?

Any help is appreciated.

Thanks.

**Comment by Nic-Ma** [comment]
Hi @yeeseng ,

Thanks for your interest here.
Actually, MONAI has 2 sets of transforms: array based and dict based.
You can check this doc for more details:
https://docs.monai.io/en/latest/highlights.html#transforms-support-both-dictionary-and-array-format-data
For your problem, you can make your data as a dict with keys and apply the dict-based transforms, example:
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/torch/unet_training_dict.py
Or use array-based transforms for images and labels, example:
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/torch/unet_training_array.py

Thanks.
